comp/core/autodiscovery/common/utils/annotations.go

```diff
@@ -81,11 +81,17 @@ func extractCheckTemplatesFromMap(key string, input map[string]string, prefix st
 
 // extractLogsTemplatesFromMap returns the logs configuration from a given map,
 // if none are found return an empty list.
-func extractLogsTemplatesFromMap(key string, input map[string]string, prefix string) ([]integration.Config, error) {
+func extractLogsTemplatesFromMap(configs []integration.Config, key string, input map[string]string, prefix string) ([]integration.Config, error) {
 	value, found := input[prefix+logsConfigPath]
 	if !found {
 		return []integration.Config{}, nil
 	}
+	logCheckName := ""
+	if len(configs) >= 1 {
```

Comment:
It's possible to have different names in different configs, and it would mean that one attached multiple integration to a single container (e.g. `redis` + `nginx`). We expect we won't encounter this most of the time, but if it happens it means we're tagging the wrong integration name

Reply:
Thanks for clarifying. It might be helpful for future reference to add a comment in the code with what you explained in your comment.